src/lua/luaTask.c

```diff
@@ -300,64 +339,176 @@ void run_lua_interactive_cmd(Serial *serial, const char* cmd)
-                if (LUA_ERRMEM != rc) {
-                        fales = 0;
-                        continue;
-                }
+bool lua_task_start()
+{
+        if (!is_init(false) || is_runtime_active())
+                return false;
 
-                /* If here then LUA failed to alloc RAM. */
-                if (++fales < 3)
-                        continue;
+        get_lock();
+        bool ok = false;
 
-                pr_warning("[lua] LUA Memory Failure.  Resetting LUA\r\n");
-                terminate_lua();
-                initialize_lua();
-                fales = 0;
+        /* Initialize the Lua runtime here */
+        state.lua_runtime = setup_lua_state();
+        if (!is_runtime_active()) {
+                pr_warning(_LOG_PFX "Failed to create lua runtime\r\n");
+                goto done;
         }
+
+        pr_info(_LOG_PFX "Starting Lua Task\r\n");
+        ok = pdPASS == xTaskCreate(lua_task, (signed portCHAR *) "Lua Task",
+                                   LUA_STACK_SIZE, state.lua_runtime,
+                                   state.priority, &state.task_handle);
+
+        if (!ok) {
+                pr_warning(_LOG_PFX "Failed to start Lua Task\r\n");
+                state.task_handle = NULL; /* Just to be sure */
+        }
+done:
+        release_lock();
+        return ok;
 }
 
-void startLuaTask(int priority)
+/**
+ * Checks if we need to bypass the Lua script at the user's request.
+ * @return True if asked to do so, false otherwise.
+ */
+static bool user_bypass_check()
 {
-        vSemaphoreCreateBinary(xLuaLock);
-        lua_run_state = LUA_DISABLED;
+        /* Only check if reset cause was a watchdog event */
+        if (!watchdog_is_watchdog_reset())
+                return false;
+
+        pr_info(_LOG_PFX "Runtime bypass check triggered...\r\n");
+        bool bypass = false;
+        led_disable(LED_ERROR);
+
+	const int flash_count =
+                LUA_BYPASS_DELAY_SEC * 1000 / LUA_FLASH_DELAY_MS;
+        for (size_t i = 0; i < flash_count; ++i) {
+                delayMs(LUA_FLASH_DELAY_MS);
+                led_toggle(LED_ERROR);
+                if (GPIO_is_button_pressed()) {
+                        bypass = true;
+                        break;
+                }
+        }
+
+        led_disable(LED_ERROR);
+        pr_info_bool_msg(_LOG_PFX "Bypass requested: ", bypass);
+
+        return bypass;
+}
+
+/**
+ * Called when we first setup the system.  Should only ever be called
+ * once.
+ */
+bool lua_task_init(const int priority)
+{
+        /* Only init once */
+        if (is_init(true)) {
+                pr_error(_LOG_PFX "Already initialized\r\n");
+                return false;
+        }
+
+        /*
+         * Always set initial state here.  Needed because we
+         * may kill this task later.  Setting the lock is what we use
+         * to determine that we are initialized.
+         */
+        vSemaphoreCreateBinary(state.semaphore);
```

Comment:
Your first commit was to replace this with a mutex to allow for priority inheritance.  Was it intentional that you changed it back?